web/src/components/core/PasswordInput.test.jsx

```diff
@@ -0,0 +1,171 @@
+  });
+
+  it("can be disabled", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        isDisabled={true}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    const visibilityButton = screen.getByRole("button");
+
+    expect(inputField).toBeDisabled();
+    expect(visibilityButton).toBeDisabled();
+  });
+
+  it("applies autoFocus behavior correctly", () => {
+    plainRender(
+      <PasswordInput
+        autoFocus
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+
+    expect(document.activeElement).toBe(inputField);
+  });
+
+  it("displays helper text for invalid input", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        helperTextInvalid="Invalid input"
+        validated="error"
+      />
+    );
+    const helperText = screen.getByText("Invalid input");
+
+    expect(helperText).toBeInTheDocument();
+  });
+
+  it("calls onBlur event handler", () => {
+    const onBlurMock = jest.fn();
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("User password")}
+        onBlur={onBlurMock}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+
+    fireEvent.blur(inputField);
+    expect(onBlurMock).toHaveBeenCalled();
+  });
+
+  it("ensures accessibility attributes are set correctly", () => {
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("Password")}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    const visibilityButton = screen.getByLabelText("Password visibility button");
+
+    expect(inputField).toHaveAttribute("aria-label", "Password");
+    expect(visibilityButton).toHaveAttribute("aria-label", "Password visibility button");
+  });
+
+  it("handles setting and changing the input value", async () => {
+    let inputValue = "";
+    const handleChange = (value) => {
+      inputValue = value;
+    };
+
+    plainRender(
+      <PasswordInput
+        id="password"
+        name="password"
+        label={_("Password")}
+        fieldId="password"
+        ariaLabel={_("Password")}
+        value={inputValue}
+        onChange={handleChange}
+      />
+    );
+    const inputField = screen.getByLabelText("Password");
+    
+    await fireEvent.change(inputField, { target: { value: "newpassword" }});
```

Comment:
The spell checking is complaining because "newpassword" does not look as a valid word for it. You can use "new-password", "secret" or so, but if this test example is finally deleted the complaint will go as well.